src/main/scala/Configs.scala

```diff
@@ -50,6 +50,21 @@ class WithNBlockDeviceTrackers(n: Int) extends Config((site, here, up) => {
   }
 })
 
-class WithTSI extends Config((site, here, up) => {
-  case SerialKey => true
+class WithSerialTSI extends Config((site, here, up) => {
+  case SerialTSIKey => Some(SerialTSIParams())
+})
+
+class WithNoSerialTSI extends Config((site, here, up) => {
+  case SerialTSIKey => None
+})
+
+class WithDefaultSerialTL extends Config((site, here, up) => {
+  case SerialTLKey => Some(SerialTLParams(
+    memParams = MemoryPortParams(MasterPortParams(
+      base = BigInt("80000000", 16),
```

Comment:
We can't use only this for off-chip memory, we still have to support backing AXI4 memory for FireSim.
As a result, we can't use `ExtMem`, since setting `ExtMem` enables the backing AXI4 memory.

Reply:
Are you saying I can't model a chip that only has `SerialTL` as its backing memory in FireSim?

Reply:
You can. Sorry what I typed was confusing. 

We can't use the `ExtMem` key here, because setting `ExtMem` enables the AXI4 memory.

Reply:
Ah I see this is a rocket-chip constraint 😢

Reply:
Is that new? We used to use ExtMem for this purpose

Reply:
You didn't include the `CanHaveMasterAXI4MemPort` trait. Since we now have a "universal" digital system, we can't overload the purpose of `ExtMem` anymore.

Reply:
Can we add a `JKCantHaveMasterAXI4MemPort` trait?

Reply:
In all seriousness, this is probably fine since we are deprecating this, but maybe the new thing (if/when it ever exists) should speak AXI4 and connect to the Master AXI4 Mem Port (which makes it look like a DDR controller).

Reply:
That would be cool.

Are we deprecating this though? It seems to be sufficiently simple for the very limited use-case of a TL-UL backing memory.

Reply:
Oh, I guess there's no need to deprecate this utility, but for chips, at least, you probably want this to be the primary backing memory (ExtMem). If you have multiple backing memory interfaces (e.g. LBWIF + HBWIF), you have two options:
1. a switch component, which we found out through EAGLE is kind of annoying to get working with diplomacy
2. double the ExtMem space and map half to each port, which adds a bit to every address/tag/etc. and also requires you to compile different binaries for each port type

If this is just a secondary interface that isn't intended to be the main backing memory, then 2 works fine, I think.

Reply:
I think in the chipyard meeting, we want this to be the primary backing memory for chips. We currently don't have immediate plans to integrate another backing memory interface.

Reply:
👍 We don't need to integrate the switcher yet since there is nothing to switch to.
I personally don't like the compile everything twice approach but we don't need to cross that bridge now anyways.